pkg/cloudprovider/provider/vsphere/helper.go

```diff
@@ -69,10 +69,29 @@ func createClonedVM(ctx context.Context, vmName string, config *Config, session
 		targetVMFolder = datacenterFolders.VmFolder
 	}
 
+	var targetResourcePool *object.ResourcePool
+	if config.ResourcePool != "" {
+		targetResourcePool, err = session.Finder.ResourcePool(ctx, config.ResourcePool)
+		if err != nil {
+			return nil, fmt.Errorf("failed to get target resourcepool: %v", err)
+		}
+	} else {
```

Comment:
@maxilampert I don't disagree on the principle of defaulting to the root pool associated to the  `ClusterComputeResource`, but this still changes a bit the behaviour. With a regular VM the same pool is used if not otherwise specified. Don't know if someone is relying on this but it is better to stay on the safe side.

What I propose is to use this logic only in case of templates. The code snippet below could be used to obtain the info:

```
	isTemplate, err = vm.IsTemplate(ctx)
	if err != nil {
		return nil, fmt.Errorf("failed to get vm template property: %v", err)
	}

```

Reply:
@irozzo-1A I restored the old default logic for cloned VMs and moved them into a func to clean up the  `createCloneVM` func. All of the values can be overwritten by the `resourcePool` field, for templates it defaults to the cluster default pool. I think that restores the default behaviour and allows the most possible flexibility for the other options.

In regards to testing I wonder how the infrastructure look like for the E2E tests what ResourcePools to use and what use cases should be verified?